### PR TITLE
dnsdist: Update ChangeLog and secpoll for 1.8.4

### PR DESCRIPTION
**Pull request by rgacogne**
### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/dnsdistdist/docs/changelog.rst [hunk 1]
```diff
@@ -1,6 +1,31 @@
...
+  :version: 1.8.4
+  :released: 20th of September 2024
+
+  Please review the :doc:`Upgrade Guide <../upgrade_guide>` before upgrading from versions < 1.8.x.
+
+  .. change::
+    :tags: Bug Fixes
+    :pullreq: 14467
+
+    Fix a compilation issue with clang by switching to pdns::views::UnsignedCharView
```
Comment (Habbie) [inline pdns/dnsdistdist/docs/changelog.rst:14]:
```suggestion
    Fix a compilation issue with clang by switching to ``pdns::views::UnsignedCharView``
```

File: pdns/dnsdistdist/docs/changelog.rst [hunk 2]
```diff
@@ -1,6 +1,31 @@
...
+    :pullreq: 14680
+    :tickets: 14562
+
+    Fix build with boost 1.86.0
+
+  .. change::
+    :tags: Bug Fixes, DNS over TLS
+    :pullreq: 14679
+
+    Prevent a data race in incoming DNS over TLS connections by storing the `OpenSSLTLSIOCtx` in the connection
```
Comment (Habbie) [inline pdns/dnsdistdist/docs/changelog.rst:27]:
```suggestion
    Prevent a data race in incoming DNS over TLS connections by storing the ``OpenSSLTLSIOCtx`` in the connection
```

**Review by Habbie — CHANGES_REQUESTED** [review]
two nits, otherwise good, agreed on all the PRs left out of this

File: pdns/dnsdistdist/docs/changelog.rst (hunk 2 above)
Comment (rgacogne) [reply pdns/dnsdistdist/docs/changelog.rst]: Good catch, thanks! I did check the rendering in the PDF but apparently a single backtick somehow works there :man_shrugging:

Comment (Habbie) [reply pdns/dnsdistdist/docs/changelog.rst]: I see it in italics in both PDF and HTML with the single ticks, which is not terrible, but not what I'd choose

Comment (rgacogne) [reply pdns/dnsdistdist/docs/changelog.rst]: For the record, I was wrong! Single tick renders in italics, double tick as a "code" style. Note to self: double-check that the PDF was actually re-generated next time.

**Comment by rgacogne** [comment]
Rebased on master to get the new version of the mssql image in our CI.

**Review by aerique — APPROVED** [review]

**Pull request merged**